examples/cpp/main.cpp

```diff
@@ -231,6 +231,56 @@ void TestThirdParty() {
       false, "image");
 }
 
+void TestClassId() {
+  Engine engine(
+      "###element\n"
+      "##.ads\n"
+      "##.element\n"
+      "###ads > #element\n"
+      "##a[href^=\"test.com\"]\n"
+      "###block\n"
+      "###block + .child\n"
+  );
+  std::vector<std::string> classes = std::vector<std::string>();
+  std::vector<std::string> ids = std::vector<std::string>();
+  std::vector<std::string> exceptions = std::vector<std::string>();
+  std::string stylesheet = engine.classIdStylesheet(classes, ids, exceptions);
+  assert(stylesheet == "");
+
+  classes = std::vector<std::string>({"ads", "no-ads"});
+  ids = std::vector<std::string>({"element"});
+  exceptions = std::vector<std::string>();
+  stylesheet = engine.classIdStylesheet(classes, ids, exceptions);
+  assert(stylesheet == ".ads,#element{display:none !important;}");
+
+  classes = std::vector<std::string>({"element", "a"});
+  ids = std::vector<std::string>({"block", "ads", "a"});
+  exceptions = std::vector<std::string>({"#block"});
+  stylesheet = engine.classIdStylesheet(classes, ids, exceptions);
+  assert(stylesheet == ".element,#block + .child,#ads > #element{display:none !important;}");
+}
+
+void TestHostnameCosmetics() {
+  Engine engine(
+      "a.com###element\n"
+      "b.com##.ads\n"
+      "##.block\n"
+      "a.com#@#.block\n"
+      "##a[href=\"b.com\"]\n"
+      "b.*##div:style(background: #fff)\n"
+  );
+
+  std::string a_resources = engine.hostnameCosmeticResources("a.com");
+  std::string a_order1("{\"hide_selectors\":[\"a[href=\\\"b.com\\\"]\",\"#element\"],\"style_selectors\":{},\"exceptions\":[\".block\"],\"injected_script\":\"\"}");
+  std::string a_order2("{\"hide_selectors\":[\"#element\",\"a[href=\\\"b.com\\\"]\"],\"style_selectors\":{},\"exceptions\":[\".block\"],\"injected_script\":\"\"}");
+  assert(a_resources == a_order1 || a_resources == a_order2);
```

Comment:
Just curious, order is not guaranteed to be deterministic?

Reply:
Nope, these are pulled straight from `HashMap`s in the standard Rust library. Order of iteration is non-deterministic. Could probably choose another data structure if that's desired.

Reply:
Non-deterministic is ok, just checking. Thanks.